lib/memo_wise.rb

```diff
@@ -236,6 +238,20 @@ def self.create_memo_wise_state!(obj)
   #
   def self.prepended(target) # rubocop:disable Metrics/PerceivedComplexity
     class << target
+      # Allocator to setup memoization state before
```

Comment:
I really wasn't sure of the best way to document this method. Suggestions welcome.

Reply:
Love it! Can we also add links to:

Documentation of `Class#allocate`
- https://ruby-doc.org/core-2.7.2/Class.html#method-i-allocate

Relevant call to `Class#allocate` in ActiveRecord 5
- https://github.com/rails/rails/blob/a395c3a6af1e079740e7a28994d77c8baadd2a9d/activerecord/lib/active_record/persistence.rb#L411